client.go

```diff
@@ -5,16 +5,19 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"net"
 	"net/http"
 	"net/url"
 	"path"
+	"time"
 )
 
 // ClientConfig object used for client creation
 type ClientConfig struct {
 	APIHost    string
 	APIKey     string
 	OAuthToken string
+	HTTPClient *http.Client
```

Comment:
I'd put timeout configuration fields as `int` in the `ClientConfig` struct, instead of having the object instance here.
I think `*http.Client` instance belongs to `Client` struct otherwise.

Reply:
I agree. I wasn't very satisfied with just adding the client here, but wasn't sure if there are other customizations to be made on an `http.Client`. I've only ever changed the timeouts.